### ISSUE TITLE
[Bug Report] Union of xpath in performerSearch results in wrong performer URL

### ISSUE THREAD

**Issue by MrX292** [stashapp/stash#444]
**Describe the bug**
Using | to join xpath for Performersearch in the namefield and the url results in wrong Performer Url with not complete Names.

**To Reproduce**
1. Using the iafd scraper search for part of the name, for example david or tiff
2. pick something on down the list
3. Returns with different Performer

The not joined one works.

**Srapers**
joined
```
name: iafd
performerByName:
  action: scrapeXPath
  queryURL: http://www.iafd.com/results.asp?searchtype=comprehensive&searchstring={}
  scraper: performerSearch
performerByURL:
  - action: scrapeXPath
    url: 
      - http://www.iafd.com
    scraper: performerScraper

xPathScrapers:
  performerSearch:
    performer:
      Name: //table[@id='tblFem']//td[a[img]]/following-sibling::td[1]/a/text()|//table[@id='tblMal']//td[a[img]]/following-sibling::td[1]/a/text()
      URL: 
        selector: //table[@id='tblFem']//td[a[img]]/following-sibling::td[1]/a/@href|//table[@id='tblMal']//td[a[img]]/following-sibling::td[1]/a/@href
        replace:
          - regex: ^
            with: http://www.iafd.com
  
  performerScraper:
    performer:
      Name: //h1
      Gender: 
        selector: //form[@id='correct']/input[@name='Gender']/@value
        replace: 
          - regex: f
            with: "Female"
          - regex: m
            with: "Male"
      URL: //head/meta[@property="og:url"]/@content
      Twitter: 
        selector: //p[@class='biodata']/a[contains(text(),"http://twitter.com/")]/@href
        replace:
          - regex: http://twitter.com/
            with: ""
      Instagram: 
        selector: //p[@class='biodata']/a[contains(text(),"http://instagram.com/")]/@href
        replace:
          - regex: http://instagram.com/
            with: ""
      Birthdate:
        selector: //div[p[text()='Birthday']]/p[@class='biodata']/a[contains(@href, '/calendar.')]/text()
        # reference date is: 2006/01/02
        parseDate: January 2, 2006
      Ethnicity: 
        selector: //div[p[text()='Ethnicity']]/p[@class='biodata'][1]
        replace: 
          - regex: Asian
            with: "asian"
          - regex: Caucasian
            with: "white"
          - regex: Black
            with: "black"
          - regex: Latin
            with: "hispanic"
      Country: //div/p[text()='Nationality']/following-sibling::p[1]/text()
      Height: 
        selector: //div/p[text()='Height']/following-sibling::p[1]/text()
        replace: 
          - regex: \d+\D+\d+\D+
            with: ""
          - regex:  \D+$
            with: ""
      Measurements: //div/p[text()='Measurements']/following-sibling::p[1]/text()
      CareerLength: 
        selector: //div/p[@class='biodata'][contains(text(),"Started around")]/text()
        replace: 
          - regex: (\D+\d\d\D+)$
            with: ""     
      Aliases: //div[p[@class='bioheading']]//div[@class='biodata']/text()
      Tattoos: //div/p[text()='Tattoos']/following-sibling::p[1]/text()
      Piercings: //div/p[text()='Piercings']/following-sibling::p[1]/text()
      Image:
        selector: //div[@id='headshot']/img/@src
```
seperat
```
name: iafdFemale
performerByName:
  action: scrapeXPath
  queryURL: http://www.iafd.com/results.asp?searchtype=comprehensive&searchstring={}
  scraper: performerSearch
performerByURL:
  - action: scrapeXPath
    url: 
      - http://www.iafd.com
    scraper: performerScraper

xPathScrapers:
  performerSearch:
    performer:
      Name: //table[@id='tblFem']//td[a[img]]/following-sibling::td[1]/a/text()
      URL: 
        selector: //table[@id='tblFem']//td[a[img]]/following-sibling::td[1]/a/@href
        replace:
          - regex: ^
            with: http://www.iafd.com
  
  performerScraper:
    performer:
      Name: //h1
      Gender: 
        selector: //form[@id='correct']/input[@name='Gender']/@value
        replace: 
          - regex: f
            with: "Female"
          - regex: m
            with: "Male"
      URL: //head/meta[@property="og:url"]/@content
      Twitter: 
        selector: //p[@class='biodata']/a[contains(text(),"http://twitter.com/")]/@href
        replace:
          - regex: http://twitter.com/
            with: ""
      Instagram: 
        selector: //p[@class='biodata']/a[contains(text(),"http://instagram.com/")]/@href
        replace:
          - regex: http://instagram.com/
            with: ""
      Birthdate:
        selector: //div[p[text()='Birthday']]/p[@class='biodata']/a[contains(@href, '/calendar.')]/text()
        # reference date is: 2006/01/02
        parseDate: January 2, 2006
      Ethnicity: 
        selector: //div[p[text()='Ethnicity']]/p[@class='biodata'][1]
        replace: 
          - regex: Asian
            with: "asian"
          - regex: Caucasian
            with: "white"
          - regex: Black
            with: "black"
          - regex: Latin
            with: "hispanic"
      Country: //div/p[text()='Nationality']/following-sibling::p[1]/text()
      Height: 
        selector: //div/p[text()='Height']/following-sibling::p[1]/text()
        replace: 
          - regex: \d+\D+\d+\D+
            with: ""
          - regex:  \D+$
            with: ""
      Measurements: //div/p[text()='Measurements']/following-sibling::p[1]/text()
      CareerLength: 
        selector: //div/p[@class='biodata'][contains(text(),"Started around")]/text()
        replace: 
          - regex: (\D+\d\d\D+)$
            with: ""     
      Aliases: //div[p[@class='bioheading']]//div[@class='biodata']/text()
      Tattoos: //div/p[text()='Tattoos']/following-sibling::p[1]/text()
      Piercings: //div/p[text()='Piercings']/following-sibling::p[1]/text()
      Image:
        selector: //div[@id='headshot']/img/@src

```

**Desktop (please complete the following information):**
 - win 10
 - firefox
 - b3e8d1e

**Comment by WithoutPants** [comment]
Haven't been able to reproduce this so far. Using the joined config, queried for `david` and chose `Jutta David` - which is the fifth result on the list - then clicked Scrape. Successfully scraped the correct performer.

Can you provide exact steps to reproduce including exact query string and chosen performer?

**Comment by MrX292** [comment]
after Davida it returns the previous href.
Checked the search on iafd [http://www.iafd.com/results.asp?searchtype=comprehensive&searchstring=david](url) and there are two Davida on the list but with
`//table[@id='tblFem']//td[a[img]]/following-sibling::td[1]/a/text()|//table[@id='tblMal']//td[a[img]]/following-sibling::td[1]/a/text()` it only shows one.
but when i change
```
       Name: //table[@id='tblFem' or @id='tblMal']//td[a[img]]/following-sibling::td[1]/a/text()
      URL: 
        selector: //table[@id='tblFem' or @id='tblMal']//td[a[img]]/following-sibling::td[1]/a/@href
        replace:
          - regex: ^
            with: http://www.iafd.com
```
i get both

**Comment by WithoutPants** [comment]
I was finally able to reproduce this bug. This issue is occurring because of the two instances of the same performer name in the IAFD results, and a bug with the `htmlquery` external library which drops elements with the same inner text when using `text()` and the `|` character.

I have reported the issue here: https://github.com/antchfx/htmlquery/issues/25

I have confirmed that, in addition to the alternative xpath you mentioned above, if you remove the `text()` part of the original `Name` selector, then it also works correctly.
